[0.9.0-BETA] Unknown plugin message need more a user-friendly.

This isn't a bug. a feature request. 
0.9.0-BETA output `Error: JRuby runtime is not loaded successfully.` if the specified plugin does not found.
I think it isn't a user friendly message.
Is it possible to output `GuessPlugin 'hogehoge' is not found` like the 0.8.X ?


```
embulk-0.8.39 guess -g hogehoge guess.yml
...
Error: GuessPlugin 'hogehoge' is not found.
Unknown guess plugin 'hogehoge'. embulk/guess/hogehoge.rb is not installed. Run 'embulk gem search -rd embulk-guess' command to find plugins.
```


```
embulk guess -g hogehoge guess.yml

Caused by: org.jruby.embed.InvokeFailedException: org.jruby.proxy.org.embulk.config.ConfigException$Proxy1: Unknown guess plugin 'hogehoge'. embulk/guess/hogehoge.rb is not installed. Run 'embulk gem search -rd embulk-guess' command to find plugins.

...
...

Error: JRuby runtime is not loaded successfully.
```


### complete log

```
embulk guess -g hogehoge guess.yml
2018-01-29 10:25:22.454 +0900: Embulk v0.9.0-BETA

********************************** INFORMATION **********************************
  Join us! Embulk-announce mailing list is up for IMPORTANT annoucement such as
    compatibility-breaking changes and key feature updates.
  https://groups.google.com/forum/#!forum/embulk-announce
*********************************************************************************

2018-01-29 10:25:24.741 +0900 [INFO] (main): Gem's home and path are set by default: "/Users/hsato/.embulk/lib/gems"
2018-01-29 10:25:25.437 +0900 [INFO] (0001:guess): Listing local files at directory '.' filtering filename by prefix 'test.json'
2018-01-29 10:25:25.438 +0900 [INFO] (0001:guess): "follow_symlinks" is set false. Note that symbolic links to directories are skipped.
2018-01-29 10:25:25.441 +0900 [INFO] (0001:guess): Loading files [test.json]
2018-01-29 10:25:25.449 +0900 [INFO] (0001:guess): Try to read 32,768 bytes from input source
2018-01-29 10:25:25.494 +0900 [INFO] (0001:guess): Loaded plugin embulk (0.9.0.beta)
2018-01-29 10:25:25.509 +0900 [INFO] (0001:guess): Loaded plugin embulk (0.9.0.beta)
2018-01-29 10:25:25.527 +0900 [INFO] (0001:guess): Loaded plugin embulk (0.9.0.beta)
2018-01-29 10:25:25.536 +0900 [INFO] (0001:guess): Loaded plugin embulk (0.9.0.beta)
org.embulk.jruby.JRubyNotLoadedException: JRuby runtime is not loaded successfully.
	at org.embulk.jruby.ScriptingContainerDelegate.callMethod(ScriptingContainerDelegate.java:573)
	at org.embulk.jruby.JRubyPluginSource.newPlugin(JRubyPluginSource.java:65)
	at org.embulk.plugin.PluginManager.newPluginWithoutWrapper(PluginManager.java:48)
	at org.embulk.plugin.PluginManager.newPlugin(PluginManager.java:27)
	at org.embulk.spi.ExecSession.newPlugin(ExecSession.java:149)
	at org.embulk.spi.Exec.newPlugin(Exec.java:64)
	at org.embulk.exec.GuessExecutor$GuessParserPlugin.run(GuessExecutor.java:209)
	at org.embulk.spi.FileInputRunner.run(FileInputRunner.java:140)
	at org.embulk.exec.GuessExecutor$2.run(GuessExecutor.java:150)
	at org.embulk.spi.FileInputRunner$RunnerControl$1$1.run(FileInputRunner.java:112)
	at org.embulk.exec.GuessExecutor$GuessParserPlugin.transaction(GuessExecutor.java:195)
	at org.embulk.spi.FileInputRunner$RunnerControl$1.run(FileInputRunner.java:107)
	at org.embulk.spi.util.Decoders$RecursiveControl.transaction(Decoders.java:68)
	at org.embulk.spi.util.Decoders.transaction(Decoders.java:29)
	at org.embulk.spi.FileInputRunner$RunnerControl.run(FileInputRunner.java:105)
	at org.embulk.exec.BufferFileInputPlugin.transaction(BufferFileInputPlugin.java:21)
	at org.embulk.spi.FileInputRunner.transaction(FileInputRunner.java:62)
	at org.embulk.exec.GuessExecutor.guessParserConfig(GuessExecutor.java:145)
	at org.embulk.exec.GuessExecutor.guessParserConfig(GuessExecutor.java:126)
	at org.embulk.spi.FileInputRunner.guess(FileInputRunner.java:85)
	at org.embulk.exec.GuessExecutor.doGuess(GuessExecutor.java:103)
	at org.embulk.exec.GuessExecutor.access$000(GuessExecutor.java:33)
	at org.embulk.exec.GuessExecutor$1.run(GuessExecutor.java:82)
	at org.embulk.exec.GuessExecutor$1.run(GuessExecutor.java:79)
	at org.embulk.spi.Exec.doWith(Exec.java:23)
	at org.embulk.exec.GuessExecutor.guess(GuessExecutor.java:79)
	at org.embulk.EmbulkEmbed.guess(EmbulkEmbed.java:138)
	at org.embulk.EmbulkRunner.guessInternal(EmbulkRunner.java:202)
	at org.embulk.EmbulkRunner.guess(EmbulkRunner.java:59)
	at org.embulk.cli.EmbulkRun.runSubcommand(EmbulkRun.java:551)
	at org.embulk.cli.EmbulkRun.run(EmbulkRun.java:95)
	at org.embulk.cli.Main.main(Main.java:26)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.embulk.jruby.ScriptingContainerDelegate.callMethod(ScriptingContainerDelegate.java:569)
	... 31 more
Caused by: org.jruby.embed.InvokeFailedException: org.jruby.proxy.org.embulk.config.ConfigException$Proxy1: Unknown guess plugin 'hogehoge'. embulk/guess/hogehoge.rb is not installed. Run 'embulk gem search -rd embulk-guess' command to find plugins.
	at org.jruby.embed.internal.EmbedRubyObjectAdapterImpl.call(EmbedRubyObjectAdapterImpl.java:322)
	at org.jruby.embed.internal.EmbedRubyObjectAdapterImpl.callMethod(EmbedRubyObjectAdapterImpl.java:159)
	at org.jruby.embed.ScriptingContainer.callMethod(ScriptingContainer.java:1459)
	... 36 more
Caused by: org.jruby.proxy.org.embulk.config.ConfigException$Proxy1: Unknown guess plugin 'hogehoge'. embulk/guess/hogehoge.rb is not installed. Run 'embulk gem search -rd embulk-guess' command to find plugins.
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.jruby.javasupport.proxy.JavaProxyConstructor.newInstanceImpl(JavaProxyConstructor.java:119)
	at org.jruby.javasupport.proxy.JavaProxyConstructor.newInstance(JavaProxyConstructor.java:205)
	at org.jruby.javasupport.Java$JCreateMethod.call(Java.java:618)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)
	at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodN.call(JavaMethod.java:739)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:278)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:79)
	at org.jruby.java.proxies.ConcreteJavaProxy$InitializeMethod.call(ConcreteJavaProxy.java:48)
	at org.jruby.ir.runtime.IRRuntimeHelpers.instanceSuper(IRRuntimeHelpers.java:984)
	at org.jruby.ir.instructions.InstanceSuperInstr.interpret(InstanceSuperInstr.java:69)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:360)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:74)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:84)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:179)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:165)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:328)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:165)
	at org.jruby.RubyClass.newInstance(RubyClass.java:1001)
	at org.jruby.RubyClass$INVOKER$i$newInstance.call(RubyClass$INVOKER$i$newInstance.gen)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)
	at org.jruby.java.proxies.ConcreteJavaProxy$NewMethod.call(ConcreteJavaProxy.java:158)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:318)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:155)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:315)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:74)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:84)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:179)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:165)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:153)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:315)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:74)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:90)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:214)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:200)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:208)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:181)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:324)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:74)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:109)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:95)
	at org.jruby.RubyClass.finvoke(RubyClass.java:522)
	at org.jruby.RubyBasicObject.send19(RubyBasicObject.java:1684)
	at org.jruby.RubyBasicObject$INVOKER$i$send19.call(RubyBasicObject$INVOKER$i$send19.gen)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:77)
	at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:432)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:360)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:74)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:109)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:95)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:174)
	at org.jruby.RubyClass.finvoke(RubyClass.java:899)
	at org.jruby.runtime.Helpers.invoke(Helpers.java:430)
	at org.jruby.embed.internal.EmbedRubyObjectAdapterImpl.callEachType(EmbedRubyObjectAdapterImpl.java:356)
	at org.jruby.embed.internal.EmbedRubyObjectAdapterImpl.call(EmbedRubyObjectAdapterImpl.java:309)
	... 38 more

Error: JRuby runtime is not loaded successfully.
```

--- a/embulk-core/src/main/java/org/embulk/jruby/ScriptingContainerDelegate.java
+++ b/embulk-core/src/main/java/org/embulk/jruby/ScriptingContainerDelegate.java
@@ -526,8 +526,18 @@ public final class ScriptingContainerDelegate {
         }
         try {
             return this.method_callMethod_ALObject.invoke(this.scriptingContainer, receiver, methodName, args);
-        } catch (IllegalAccessException | InvocationTargetException ex) {
+        } catch (IllegalAccessException ex) {
             throw new JRubyNotLoadedException(ex);
+        } catch (InvocationTargetException ex) {
+            if (isCausedByInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex.getCause());
+            }
+            throw new JRubyNotLoadedException(ex);
+        } catch (RuntimeException ex) {
+            if (isInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex);
+            }
+            throw ex;
         }
     }
 
@@ -550,10 +560,18 @@ public final class ScriptingContainerDelegate {
             final T returnValue = (T) (this.method_callMethod_LClass.invoke(
                     this.scriptingContainer, receiver, methodName, returnType));
             return returnValue;
-        } catch (IllegalAccessException | InvocationTargetException ex) {
+        } catch (IllegalAccessException ex) {
+            throw new JRubyNotLoadedException(ex);
+        } catch (InvocationTargetException ex) {
+            if (isCausedByInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex.getCause());
+            }
             throw new JRubyNotLoadedException(ex);
         } catch (RuntimeException ex) {
-            throw wrapJRubyInvokeFailedException(ex);
+            if (isInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex);
+            }
+            throw ex;
         }
     }
 
@@ -569,10 +587,18 @@ public final class ScriptingContainerDelegate {
             final T returnValue = (T) (this.method_callMethod_LObject_LClass.invoke(
                     this.scriptingContainer, receiver, methodName, singleArg, returnType));
             return returnValue;
-        } catch (IllegalAccessException | InvocationTargetException ex) {
+        } catch (IllegalAccessException ex) {
+            throw new JRubyNotLoadedException(ex);
+        } catch (InvocationTargetException ex) {
+            if (isCausedByInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex.getCause());
+            }
             throw new JRubyNotLoadedException(ex);
         } catch (RuntimeException ex) {
-            throw wrapJRubyInvokeFailedException(ex);
+            if (isInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex);
+            }
+            throw ex;
         }
     }
 
@@ -627,10 +653,18 @@ public final class ScriptingContainerDelegate {
         }
         try {
             return this.method_getProvider.invoke(this.scriptingContainer);
-        } catch (IllegalAccessException | InvocationTargetException ex) {
+        } catch (IllegalAccessException ex) {
+            throw new JRubyNotLoadedException(ex);
+        } catch (InvocationTargetException ex) {
+            if (isCausedByInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex.getCause());
+            }
             throw new JRubyNotLoadedException(ex);
         } catch (RuntimeException ex) {
-            throw wrapJRubyInvokeFailedException(ex);
+            if (isInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex);
+            }
+            throw ex;
         }
     }
 
@@ -640,10 +674,18 @@ public final class ScriptingContainerDelegate {
         }
         try {
             return this.method_put_LString_LObject.invoke(this.scriptingContainer, key, value);
-        } catch (IllegalAccessException | InvocationTargetException ex) {
+        } catch (IllegalAccessException ex) {
+            throw new JRubyNotLoadedException(ex);
+        } catch (InvocationTargetException ex) {
+            if (isCausedByInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex.getCause());
+            }
             throw new JRubyNotLoadedException(ex);
         } catch (RuntimeException ex) {
-            throw wrapJRubyInvokeFailedException(ex);
+            if (isInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex);
+            }
+            throw ex;
         }
     }
 
@@ -653,10 +695,18 @@ public final class ScriptingContainerDelegate {
         }
         try {
             return this.method_remove_LString.invoke(this.scriptingContainer, key);
-        } catch (IllegalAccessException | InvocationTargetException ex) {
+        } catch (IllegalAccessException ex) {
+            throw new JRubyNotLoadedException(ex);
+        } catch (InvocationTargetException ex) {
+            if (isCausedByInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex.getCause());
+            }
             throw new JRubyNotLoadedException(ex);
         } catch (RuntimeException ex) {
-            throw wrapJRubyInvokeFailedException(ex);
+            if (isInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex);
+            }
+            throw ex;
         }
     }
 
@@ -666,10 +716,18 @@ public final class ScriptingContainerDelegate {
         }
         try {
             return this.method_runScriptlet_LString.invoke(this.scriptingContainer, script);
-        } catch (IllegalAccessException | InvocationTargetException ex) {
+        } catch (IllegalAccessException ex) {
+            throw new JRubyNotLoadedException(ex);
+        } catch (InvocationTargetException ex) {
+            if (isCausedByInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex.getCause());
+            }
             throw new JRubyNotLoadedException(ex);
         } catch (RuntimeException ex) {
-            throw wrapJRubyInvokeFailedException(ex);
+            if (isInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex);
+            }
+            throw ex;
         }
     }
 
@@ -681,10 +739,18 @@ public final class ScriptingContainerDelegate {
         }
         try {
             return this.method_getRubyInstanceConfig.invoke(provider);
-        } catch (IllegalAccessException | InvocationTargetException ex) {
+        } catch (IllegalAccessException ex) {
+            throw new JRubyNotLoadedException(ex);
+        } catch (InvocationTargetException ex) {
+            if (isCausedByInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex.getCause());
+            }
             throw new JRubyNotLoadedException(ex);
         } catch (RuntimeException ex) {
-            throw wrapJRubyInvokeFailedException(ex);
+            if (isInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex);
+            }
+            throw ex;
         }
     }
 
@@ -696,18 +762,33 @@ public final class ScriptingContainerDelegate {
         }
         try {
             return this.method_getRuntime.invoke(provider);
-        } catch (IllegalAccessException | InvocationTargetException ex) {
+        } catch (IllegalAccessException ex) {
+            throw new JRubyNotLoadedException(ex);
+        } catch (InvocationTargetException ex) {
+            if (isCausedByInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex.getCause());
+            }
             throw new JRubyNotLoadedException(ex);
         } catch (RuntimeException ex) {
-            throw wrapJRubyInvokeFailedException(ex);
+            if (isInvokeFailedException(ex)) {
+                throw new JRubyInvokeFailedException(ex);
+            }
+            throw ex;
         }
     }
 
-    private static RuntimeException wrapJRubyInvokeFailedException(final RuntimeException exception) {
-        if (exception.getClass().getName().equals("org.jruby.embed.InvokeFailedException")) {
-            return new JRubyInvokeFailedException(exception);
+    private static boolean isCausedByInvokeFailedException(final InvocationTargetException exception) {
+        if (exception == null) {
+            return false;
         }
-        return exception;
+        final Throwable cause = exception.getCause();
+        return cause != null && isInvokeFailedException(cause);
+    }
+
+    private static boolean isInvokeFailedException(final Throwable exception) {
+        return exception != null
+                       && exception instanceof RuntimeException
+                       && exception.getClass().getName().equals("org.jruby.embed.InvokeFailedException");
     }
 
     private final String jrubyVersion;

